packages/scalecube-microservice/tests/implementations/Microservice.spec.ts

```diff
@@ -139,5 +219,19 @@ describe('Test the creation of Microservice', () => {
       const proxy1 = ms1.createProxy({ serviceDefinition: greetingServiceDefinition2 });
       return expect(proxy1.hello(defaultUser)).resolves.toEqual({}); // will fail after implementing remoteCall
     });
+
+    test(`
+      Scenario: Fail to invoke a remote service 
+        Given   two valid Microservices under the same seedAddress
+        | Microservice | reference| definition  |
+        | ms1          | s1       |d1           |
+        | ms2          | s2       |d2           | 
+        When    creating a proxy to ms1 with definition of d3
+        And     proxy tries to invoke s2
```

Comment:
```suggestion
        And     remoteCall is performed and proxy tries to invoke s2
```

Reply:
Add a scenario where we have definition 'Foo' and reference 'Foo' & 'Ro' to a single ms.
The result will be that ms will be created without 'Ro'.
`
      Scenario: Fail to register a service, 
        Given   a service with definition and reference
        |service          |definition            |reference  |
        |greetingService  |             |hello: RequestResponse|
        |                 |greet$: RequestStream |greet$: RequestStream |

        # reference has a method that is not contained in the definition
        When    creating a Microservice with the service
        Then    Microservice will successfully created
        But Microservice instance won't contain hello method
      `